library/alloc/src/vec/mod.rs

```diff
@@ -1486,6 +1488,7 @@ impl<T, A: Allocator> Vec<T, A> {
     /// ```
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[doc(alias = "append")]
```

Comment:
I think this makes sense to add, but it is unfortunate that it will pollute results for `Vec::append`.

Reply:
Do you know if the alias or `Vec::append` will be shown first in the results?

Reply:
Also could potentially be worth adding a line to `Vec::append`'s docs like:

> Not to be confused with `Vec::push`. `Vec::append` is different from `.append` in Python.

Reply:
> Do you know if the alias or `Vec::append` will be shown first in the results?

Aliases should always be shown last in the search results, I'd say, if they are not already.

> `Vec::append` is different from `.append` in Python.

I like the idea, but I think adding references to other languages in the documentation for the Rust function is going a bit too far. Instead, I'd go for #82000, allowing a `comment` field for each alias too, where this comment can be written. Then, the search can show it is best, i.e. something like:

    #[doc(alias = "append", lang = "Python", comment =
        "Not to be confused with `Vec::push`. `Vec::append` is different from `.append` in Python."
    )]

What do you think? I can add it to the RFC.

Reply:
Sure, seems reasonable. I'll drop it from this Pr.

Reply:
> Aliases should always be shown last in the search results, I'd say, if they are not already.

That seems really strange, why do you say that? That puts them behind *very* unrelated things because of fuzzy search.

![image](https://user-images.githubusercontent.com/23638587/107890698-468c4d00-6ee8-11eb-85ed-ea2428ec6d1d.png)

Reply:
> That seems really strange, why do you say that? That puts them behind _very_ unrelated things because of fuzzy search.

Ah, yeah, there are too many results. What about before the fuzzy ones? i.e.:

- Exact matches (non-aliases)
- Exact matches (aliases)
- Fuzzy search (any, assuming it sorts by some "similarity" index)

If we get #82000, then we could even do:

- Exact matches (non-aliases)
- Exact matches (Rust aliases)
- Exact matches (non-Rust aliases)
- Fuzzy search (any, assuming it sorts by some "similarity" index)

(By the way, why is matching all that for `free`?)

Reply:
> Ah, yeah, there are too many results [*]. What about before the fuzzy ones? i.e.:

I like this. Can you open an issue for it? I think it makes a lot of sense.

> [*] Why is matching all that for free? Is it due to substrings (e.g. fREm -> fREe) or Hamming distance or something like that? I think for words < 4 in size, it should not match if at least 3 characters match. But anyway, this is off-topic...

Can you open an issue for that too? :laughing:

Reply:
Done! #82131 and #82132.